source/elements/oneMKL/source/domains/blas/imatcopy.rst

```diff
@@ -0,0 +1,299 @@
+.. SPDX-FileCopyrightText: 2022 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_blas_imatcopy:
+
+imatcopy
+========
+
+Computes an in-place scaled matrix transpose or copy operation
+using a general matrix.
+
+.. _onemkl_blas_imatcopy_description:
+
+.. rubric:: Description
+
+The ``imatcopy`` routine performs an in-place scaled
+matrix copy or transposition.
+
+The operation is defined as:
+
+.. math::
+
+      AB \leftarrow \alpha * op(AB)
```

Comment:
I think having the matrix just be `A` would be better.    Unless there is a reason to name the matrix `AB` ?

Reply:
The argument in the programming API is named `AB` and I think it makes some sense to have the math description consistent with that. That's also what's done in the [oneMKL developer reference](https://www.intel.com/content/www/us/en/develop/documentation/onemkl-developer-reference-c/top/blas-and-sparse-blas-routines/blas-like-extensions/mkl-imatcopy.html). But I can certainly change it if that's better.

Reply:
What is the purpose of using the `AB` anywhere?  I don't understand where the B part of the name comes in.   what is the intuition over @aaronjohnson's suggestion to use a basic `A` or even `X` ?

Reply:
This notation predates any of my work, I'm just trying to be consistent and I'm happy to change it for clarity.

One argument for the current notation is that we have separate `lda` and `ldb` parameters here - what should they be named if `AB` becomes `matrix`? Maybe `input_ld` and `output_ld`? (and similarly for `stride_a` and `stride_b` in the batch API). I think the current naming is aiming for some consistency with `omatcopy`, this routine does the same thing but it just so happens `A` and `B` live in the same memory so we call that memory `AB`. (not saying that's ideal, but I think that's the motivation)

Reply:
And a new person reading the spec, I would find a single letter A, B or X to be less surprising.     I had pause wondering what was special about AB.     Either a single letter OR an explanation for the two would suffice.   Thank you, Andrew.

Reply:
I've redone the `imatcopy` and `imatcopy_batch` descriptions to avoid a matrix `AB`. I'm now denoting the matrix as a mathematical object as `C` and the array/buffer that holds its memory as `matrix_in_out`. I think that distinction makes sense and everything is clearer now but please let me know what you think.